src/liballoc/sync.rs

```diff
@@ -866,12 +866,10 @@ impl<T: ?Sized> Arc<T> {
     unsafe fn drop_slow(&mut self) {
         // Destroy the data at this time, even though we may not free the box
         // allocation itself (there may still be weak pointers lying around).
-        ptr::drop_in_place(&mut self.ptr.as_mut().data);
+        ptr::drop_in_place(Self::get_mut_unchecked(self));
 
-        if self.inner().weak.fetch_sub(1, Release) == 1 {
-            acquire!(self.inner().weak);
-            Global.dealloc(self.ptr.cast(), Layout::for_value(self.ptr.as_ref()))
-        }
+        // Drop the weak ref collectively held by all strong references
+        Weak { ptr: self.ptr };
```

Comment:
```suggestion
        // Drop the weak ref collectively held by all strong references.
        drop(Weak { ptr: self.ptr });
```

Reply:
(My suggestion also added the missing `.` to the end of the sentence but whatever.^^)